README.MD

```diff
@@ -1,105 +1,200 @@
+
+The protocol defines a set of actions that together perform atomic token transfers across two blockchains using gateway contracts. A gateway for a given EIP20 token is comprised of a `EIP20Gateway` contract on origin, a corresponding `EIP20CoGateway` contract on auxiliary, and and an ERC20 utility token contract on auxiliary that mints and burns utility tokens to atomically mirror tokens staked and unstaked on the origin chain.
+
+Atomicity is achieved using a 2-phase message passing architecture between the chains. Messages are declared on the source chain, and confirmed on the target chain with Patricia Merkle proofs once the source chain is finalized. Once messages are confirmed on the target chain, they can efficiently progressed with a hashlock.
+Messages can also be reverted if they are not yet completed on the target chain.
+
+You can read [the draft of the mosaic whitepaper][mosaic whitepaper] or [the original OpenST whitepaper][openst whitepaper].
+
+## Instructions
+
+### Installation
+
+```bash
+npm install @openstfoundation/mosaic.js web3 web3-eth-accounts
 ```
+> ℹ️ Note that `web3` and `web3-eth-accounts` are peer-dependencies.
+
+### Usage
+
+#### Mosaic
 
-# Abi Bin Provider Usage
 ```js
-abiBinProvider = new Mosaic.AbiBinProvider();
-jsonInterface = abiBinProvider.getABI('Organization');
-contractBin = abiBinProvider.getBIN('Organization');
-```
+const Web3 = require('web3');
+const Mosaic = require('@openstfoundation/mosaic.js');
 
+const originWeb3 = new Web3('http://localhost:8546');
+const auxiliaryWeb3 = new Web3('http://localhost:8547');
 
+const originContractAddresses = {
+  EIP20Gateway: '0x0000000000000eip20gatewaycontractaddress',
+  ...,
+};
+const auxiliaryContractAddresses = {
+  EIP20CoGateway: '0x00000000000eip20cogatewaycontractaddress',
+  ...,
+};
 
-# Using Setup Script
-Please follow these steps in sequence.
+const originChain = new Mosaic.Chain(originWeb3, originContractAddresses);
+const auxiliaryChain = new Mosaic.Chain(auxiliaryWeb3, auxiliaryContractAddresses);
+
+const mosaic = new Mosaic(originChain, auxiliaryChain);
+```
 
-## Create setup instance
+#### Facilitator
+
+Creating a new facilitator:
 ```js
-let originWeb3 = new Web3('http://127.0.0.1:8545');
-let auxiliaryWeb3 = new Web3('http://127.0.0.1:8546');
-let helper = new ChainSetup(originWeb3, auxiliaryWeb3);
-let valueToken = "0x..."; /* EIP20 value token Contract Address deployed on origin chain */
+const facilitator = new Mosaic.Facilitator(mosaic);
 ```
 
-## Create configurations. 
-Make sure all deployer addresses, organization owner addresses and chainOwner have sufficient gas and are added to web3.eth.accounts.wallet of respective chains.
+Staking:
+```js
+const staker = '0x00000000000000000000stakeraccountaddress';
+const amount = '1000000000';
+const beneficiary = '0x000000000000000beneficiaryaccountaddress';
+const gasPrice = '1';
+const gasLimit = '1000000000';
```

Comment:
Please adjust the value of `amount`, `gasPrice` and `gasLimit` such that `amount > gasPrice * gasLimit`
This condition is not added in Facilitator.js but will fail in the contract.
Or maybe we should mention this in readme.